sdk/ai/azure-ai-client/samples/agents/sample_agents_stream_eventhandler_with_functions.py

```diff
@@ -0,0 +1,140 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+"""
+FILE: sample_agents_stream_eventhandler_with_functions.py
+
+DESCRIPTION:
+    This sample demonstrates how to use agent operations with an event handler and toolset from
+    the Azure Agents service using a synchronous client.
+
+USAGE:
+    python sample_agents_stream_eventhandler_with_functions.py
+
+    Before running the sample:
+
+    pip install azure.ai.client azure-identity
+
+    Set this environment variables with your own values:
+    AI_CLIENT_CONNECTION_STRING - the Azure AI Project connection string, as found in your AI Studio Project.
+"""
+
+import os
+from azure.ai.client import AzureAIClient
+from azure.ai.client.models import MessageDeltaChunk, MessageDeltaTextContent, RunStep, ThreadMessage, ThreadRun
+from azure.ai.client.models import AgentEventHandler
+from azure.identity import DefaultAzureCredential
+from azure.ai.client.models import FunctionTool, RequiredFunctionToolCall, SubmitToolOutputsAction
+
+from typing import Any
+
+from user_functions import user_functions
+
+
+# Create an Azure AI Client from a connection string, copied from your AI Studio project.
+# At the moment, it should be in the format "<HostName>;<AzureSubscriptionId>;<ResourceGroup>;<HubName>"
+# Customer needs to login to Azure subscription via Azure CLI and set the environment variables
+
+ai_client = AzureAIClient.from_connection_string(
+    credential=DefaultAzureCredential(),
+    conn_str=os.environ["AI_CLIENT_CONNECTION_STRING"]
+)
+
+# Or, you can create the Azure AI Client by giving all required parameters directly
+"""
+ai_client = AzureAIClient(
+    credential=DefaultAzureCredential(),
+    host_name=os.environ["AI_CLIENT_HOST_NAME"],
+    subscription_id=os.environ["AI_CLIENT_SUBSCRIPTION_ID"],
+    resource_group_name=os.environ["AI_CLIENT_RESOURCE_GROUP_NAME"],
+    workspace_name=os.environ["AI_CLIENT_WORKSPACE_NAME"],
+    logging_enable=True, # Optional. Remove this line if you don't want to show how to enable logging
+)
+"""
+
+class MyEventHandler(AgentEventHandler):
+
+    def __init__(self, functions: FunctionTool) -> None:
+        self.functions = functions
+
+    def on_message_delta(self, delta: "MessageDeltaChunk") -> None:
+        for content_part in delta.delta.content:
+            if isinstance(content_part, MessageDeltaTextContent):
+                text_value = content_part.text.value if content_part.text else "No text"
+                print(f"Text delta received: {text_value}")
+
+    def on_thread_message(self, message: "ThreadMessage") -> None:
+        print(f"ThreadMessage created. ID: {message.id}, Status: {message.status}")
+
+    def on_thread_run(self, run: "ThreadRun") -> None:
+        print(f"ThreadRun status: {run.status}")
+
+        if run.status == "failed":
```

Comment:
To execute tool calls, they need to either define toolset.   Another way is to write code below which isn't desirable.    We should encourage people to use toolset.   What do you think if we add toolset in create_agent and wrap the code below as comment so that people know the desirable options?

Reply:
we can talk about forcing the toolset, however currently I think we should offer the original OpenAI's way + convenience  (toolset). So in some other PR we can propose that.